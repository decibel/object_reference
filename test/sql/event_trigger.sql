\set ECHO none

\i test/load.sql

CREATE TABLE test_table(col int);

SELECT plan(
  0
  +2 -- initial
);

SELECT lives_ok(
  $$CREATE TEMP TABLE test_object AS SELECT object_reference.object__getsert('table column', 'test_table'::regclass, 1) AS object_id;$$
  , $$CREATE TEMP TABLE test_object AS SELECT object_reference.object__getsert('table column', 'test_table'::regclass, 1) AS object_id;$$
);
SELECT is(
  (SELECT count(*) FROM test_object)
  , 1::bigint
  , 'Exactly 1 test object record'
);

\i test/pgxntool/finish.sql

-- vi: expandtab sw=2 ts=2
