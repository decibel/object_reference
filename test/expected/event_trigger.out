\set ECHO none
1..42
ok 1 - Register schema-drop test objects
ok 2 - Create objects_view
ok 3 - Exactly 3 test view records
ok 4 - CREATE TEMP TABLE column_test AS SELECT * FROM _object_reference._object_v__for_update('table column', 'table_under_test'::regclass, 1)
ok 5 - Exactly 1 test object record
ok 6 - Create column_test_view
ok 7 - Exactly 1 test view record
ok 8 - CREATE TEMP TABLE column_filler AS SELECT * FROM _object_reference._object_v__for_update('table column', 'table_under_test'::regclass, 2)
ok 9 - Exactly 1 test object record
ok 10 - Create column_filler_view
ok 11 - Exactly 1 test view record
ok 12 - CREATE TEMP TABLE table_test AS SELECT * FROM _object_reference._object_v__for_update('table', 'table_under_test'::regclass, 0)
ok 13 - Exactly 1 test object record
ok 14 - Create table_test_view
ok 15 - Exactly 1 test view record
ok 16 - Rename column
ok 17 - Update column_test
ok 18 - table_test: verify column rename
ok 19 - test: column_test: verify column rename
ok 20 - filler: column_test: verify column rename
ok 21 - Rename schema
ok 22 - Update objects table
ok 23 - Update table_test
ok 24 - Update column_test
ok 25 - table_test: verify table rename
ok 26 - test: column_test: verify table rename
ok 27 - filler: column_test: verify table rename
ok 28 - Rename table
ok 29 - Update table_test
ok 30 - Update column_test
ok 31 - table_test: verify table rename
ok 32 - test: column_test: verify table rename
ok 33 - filler: column_test: verify table rename
ok 34 - Drop column
ok 35 - Verify column record is deleted
ok 36 - Verify table record is unchanged
ok 37 - Drop table
ok 38 - Verify table record is deleted
ok 39 - Verify filler column record is deleted
ok 40 - Verify objects still registered correctly
ok 41 - Drop schema
ok 42 - objects_view is empty
# TRANSACTION INTENTIONALLY LEFT OPEN!
